chapters/classes.tex

```diff
@@ -337,25 +337,26 @@ \subsection{Acyclic Bindings of Constants and Parameters}\label{acyclic-bindings
 \begin{itemize}
 \item
   A variable \lstinline!vc! declared with \lstinline!constant!\indexinline{constant} prefix remains constant during transient analysis, with a value that is unaffected by the initialization problem.
+  This is called a \firstuse{constant}, or \firstuse{constant!variable}\index{constant!variable}\index{component variability!constant}.
 \item
   A variable \lstinline!vc! declared with the \lstinline!parameter!\indexinline{parameter} prefix remains constant during transient analysis, with a value determined by the initialization problem.
+  This is called a \firstuse{parameter}, or \firstuse{parameter!variable}\index{parameter!variable}\index{component variability!parameter}.
 \item
-  A \emph{discrete-time} variable \lstinline!vd! has a vanishing time derivative between events.
-  Note that this is not the same as saying that \lstinline!der(vd)=0! almost everywhere,
-  as the derivative is not even defined at the events, and it is not legal
-  to apply \lstinline!der! to discrete-time variables as they are not continuous. During transient analysis the variable
-  can only change its value at event
-  instants (see \cref{events-and-synchronization}).
+  A \emph{discrete-time}\index{discrete-time!variable}\index{component variability!discrete-time} variable \lstinline!vd! is a variable that is discrete-valued (that is, not of \lstinline!Real! type) or assigned in a \lstinline!when!-clause.
+  The \lstinline!discrete!\indexinline{discrete} prefix may be used to clarify that a variable is discrete-time.
+  It has a vanishing time derivative between events.
+  Note that this is not the same as saying that \lstinline!der(vd)=0! almost everywhere, as the derivative is not even defined at the events.
+  It is not allowed to apply \lstinline!der! to discrete-time variables.
+  During transient analysis the variable can only change its value at event instants (see \cref{events-and-synchronization}).
 \item
-  A \emph{continuous-time} variable \lstinline!vn! may have a non-vanishing time
-  derivative (\lstinline!der(vn)<>0! possible) and may also
-  change its value discontinuously at any time during transient analysis
-  (see \cref{events-and-synchronization}). If there are any discontinuities the variable is
-  not differentiable.
+  A \emph{continuous-time}\index{continuous-time!variable}\index{component variability!continuous-time} variable is a \lstinline!Real! variable without any prefix that is not assigned in a \lstinline!when!-clause.
+  A continuous-time variable \lstinline!vn! may have a non-vanishing time derivative (provided \lstinline!der(vn)! is allowed this can be expressed as \lstinline!der(vn) <> 0!) and may also change its value discontinuously at any time during transient analysis (see \cref{events-and-synchronization}).
+  It may also contain a combination of these effects.
+  Note that \lstinline!der(vn)! is not allowed if there are any discontinuities, see \cref{modelica:der}.
```

Comment:
Just a thought (not really requesting change): Do we really want to repeat this rather problematic statement in view of variables affected by `\reinit`?  Maybe localizing the problem to the definition of `der` would be preferable?

This variant would avoid repeating the problematic part:
```suggestion
  Regarding existence of \lstinline!der(vn)!, see \cref{modelica:der}.
```